Source/AwsCommonRuntimeKit/crt/Utilities.swift

```diff
@@ -272,3 +272,12 @@ func withByteCursorFromStrings<Result>(
         }
     }
 }
+
+func withByteCursorArrayFromStringArray<R>(
+    _ arg: [String], _ body: (UnsafePointer<aws_byte_cursor>, Int) -> R) -> R {
+    let cursors = arg.map { aws_byte_cursor_from_c_str($0) }
```

Comment:
This is not memory-safe. $0 is a Swift string and is only valid for the scope of `aws_byte_cursor_from_c_str`. The behavior is undefined after this line because the cursor refers to a memory area not guaranteed by the Swift compiler. We will need to make a copy, and then release it using our allocator.

Reply:
yeah, you are right. i added the copy back. 
is it possible to run asan or some sort of mem checker in swift?

Reply:
We do have our tracing allocator hooked up, but I never looked into ASAN. It seems easy enough to enable: https://www.swift.org/documentation/server/guides/llvm-sanitizers.html.